src/utils/mtev_memory.c

```diff
@@ -584,17 +592,34 @@ void *mtev_memory_ck_malloc(size_t r) {
 static void
 mtev_memory_ck_free_func(void *p, size_t b, bool r,
                          void (*f)(ck_epoch_entry_t *)) {
+  (void)b;
+  if(p == NULL) return;
   struct safe_epoch *e = (p - sizeof(struct safe_epoch));
 
-  if(p == NULL) return;
-  (void)b;
   bool magic_valid = e->magic == MTEV_EPOCH_SAFE_MAGIC;
+#ifdef MTEV_MEMORY_DEBUG
```

Comment:
Should be turned on by default when ASAN is detected

Reply:
I don't think so.  This adds more memory: 128 per allocation.  I often run ASAN on bug setups where that can't be afforded.

I think it should be a runtime flag, but that will take more work

Reply:
all right